cx_Freeze/freezer.py

```diff
@@ -541,7 +541,10 @@ def _WriteModules(self, fileName, finder):
                     module.file is not None:
                 parts = module.name.split(".")
                 targetPackageDir = os.path.join(targetDir, *parts)
-                sourcePackageDir = os.path.dirname(module.file)
+                if module.file is None:
```

Comment:
This code block has a check "if ... module.file is not None"
Are you sure that this happens?
Can you put a print a see what the name of the module and report?

Reply:
I get so much errors that it's difficult to understand what is really happening. This is the trace that got me to make this fix:

Traceback (most recent call last):
  File "setup/windows_service_setup.py", line 49, in <module>
    options={'build_exe': build_exe_options})
  File "C:\<<project_path>>\scripts\venv\lib\site-packages\cx_Freeze\dist.py", line 342, in setup
    distutils.core.setup(**attrs)
  File "C:\Users\phge\AppData\Local\Programs\Python\Python37\lib\distutils\core.py", line 148, in setup
    dist.run_commands()
  File "C:\Users\phge\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "C:\Users\phge\AppData\Local\Programs\Python\Python37\lib\distutils\dist.py", line 985, in run_command
    cmd_obj.run()
  File "<<project_path>>\scripts\venv\lib\site-packages\cx_Freeze\dist.py", line 217, in run
    freezer.Freeze()
  File "<<project_path>>\scripts\venv\lib\site-packages\cx_Freeze\freezer.py", line 651, in Freeze
    self._WriteModules(fileName, self.finder)
  File "<<project_path>>\scripts\venv\lib\site-packages\cx_Freeze\freezer.py", **line 540, in _WriteModules**
    sourcePackageDir = os.path.dirname(module.file)
  File "C:\Users\phge\AppData\Local\Programs\Python\Python37\lib\ntpath.py", line 221, in dirname
    return split(p)[0]
  File "C:\Users\phge\AppData\Local\Programs\Python\Python37\lib\ntpath.py", line 183, in split
    p = os.fspath(p)
**TypeError: expected str, bytes or os.PathLike object, not NoneType**

I just tried it with the if ... is None and I was able to get the executable, and run it! I really don't know what module is causing this as I have so many. I've successfully created executables for smaller projects with less dependencies. My current project has a few more dependencies and I started to get lots of problems - I think it started when I included the package nidaqmx.

Note that I believe the else could be replaced by a simple "continue" to just skip that element.

Note that I still get a log of "ResourceWarning: unclosed file" warnings but I believe these are not related at all.

Let me know if you want me to test something else! Thanks!

Reply:
More details: this case seems to happend in numpy for these folders:
.\\venv\\lib\\site-packages\\numpy\\random\\_examples
.\\venv\\lib\\site-packages\\numpy\\random\\_examples\\cffi
.\\venv\\lib\\site-packages\\numpy\\random\\_examples\\numba
These are the module.path values when module.file is None.

Reply:
You're using the cx-freeze 6.2 or current master? If not with the current master, try it out.
If you have a C compiler:
`pip install -U git+https://github.com/marcelotduarte/cx_Freeze.git@master`

Reply:
I'm using cx-freeze 6.2.
I installed Microsoft Visual C++ build tool - apparently it was needed to compile the code. Then I tried the command above and it gives me an error.
`LINK : fatal error LNK1181: cannot open input file 'cx_Logging.lib'`
Can't find the cause or a solution. Prior to that error I also get this one for wheels:
`ERROR: Failed building wheel for cx-Freeze`
Not sure what I'm doing wrong...